harrison/timer.py

```diff
@@ -45,3 +45,39 @@ def __exit__(self, *args):
         if self._verbose:
             desc = '{}: '.format(self.description) if self.description else ''
             print '{}{:.2f} ms'.format(desc, self.elapsed_time_ms)
+
+class TimeoutError(Exception):
+    pass
+
+class TimeoutTimer(Timer):
+    '''
+    Same as Timer but if given a timeout keyword argument, it will raise a
+    TimeoutError if not exitted within that number of seconds.
```

Comment:
Yeah, that seems reasonable. So maybe `def __init__(self, *args, timeout, **kwargs)` ?